src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -11,47 +11,103 @@
+        }
+
+        return parent::processValue($value, $isRoot);
+    }
+
+    private function processDefinition(Definition $definition): Definition
+    {
+        if (!$reflector = $this->container->getReflectionClass($definition->getClass(), false)) {
+            return $definition;
         }
 
-        $autoconfiguredAttributes = $container->getAutoconfiguredAttributes();
+        $autoconfiguredAttributes = $this->container->getAutoconfiguredAttributes();
 
-        foreach ($container->getDefinitions() as $id => $definition) {
-            if (!$definition->isAutoconfigured()
-                || $definition->isAbstract()
-                || $definition->hasTag($this->ignoreAttributesTag)
-                || !($reflector = $container->getReflectionClass($definition->getClass(), false))
-            ) {
-                continue;
+        $instanceof = $definition->getInstanceofConditionals();
+        $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
+        foreach ($reflector->getAttributes() as $attribute) {
+            if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
+                $configurator($conditionals, $attribute->newInstance(), $reflector);
             }
+        }
+
+        if ($constructor = $this->getConstructor($definition, false)) {
+            $definition = $this->bindArguments($definition, $constructor);
+        }
+
+        $instanceof[$reflector->getName()] = $conditionals;
+        $definition->setInstanceofConditionals($instanceof);
+
+        return $definition;
+    }
 
-            $instanceof = $definition->getInstanceofConditionals();
-            $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
-            foreach ($reflector->getAttributes() as $attribute) {
-                if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
-                    $configurator($conditionals, $attribute->newInstance(), $reflector);
+    private function bindArguments(Definition $definition, \ReflectionFunctionAbstract $constructor): Definition
+    {
+        $bindings = $definition->getBindings();
+        foreach ($constructor->getParameters() as $reflectionParameter) {
+            $argument = null;
+            foreach ($reflectionParameter->getAttributes() as $attribute) {
+                if (!$configurator = $this->argumentConfigurators[$attribute->getName()] ?? null) {
+                    continue;
+                }
+                if ($argument) {
+                    throw new LogicException(sprintf('Cannot autoconfigure argument "$%s": More than one autoconfigurable attribute found.', $reflectionParameter->getName()));
                 }
+                $argument = $configurator($attribute->newInstance(), $reflectionParameter);
+            }
+            if ($argument) {
+                $bindings['$'.$reflectionParameter->getName()] = new BoundArgument($argument);
```

Comment:
by-name only works when only the constructor is wired, but if we target setters, we'd need a more specific selector. We might need a selector to bind by method+arg-name, e.g. `setcontainer::$foo`

Reply:
Can't we "just" set the value of the argument in the call definition actually, isn't it

Reply:
Maybe this belongs to `ResolveBindingsPass`?

Reply:
> Can't we "just" set the value of the argument in the call definition actually, isn't it

That would mean I need to duplicate logic from `ResolveBindingsPass`.

> Random thought: maybe this belongs to `ResolveBindingsPass`? (or similar)

Possibly. But I must admit that I like that `ResolveBindingsPass` does not need to care about attributes or autoconfiguration at all at the moment.